### PR TITLE
chapter 12 created

### PR DESCRIPTION
**Pull request by agustinaGallo97**
## :link: Jira Tracker reference:
* [AP - Chapter 12] Fragment navigation - https://androidtraining.atlassian.net/browse/AT-52

## Description:
Chapter 12 created

## :camera_flash: UI preview
ezgif com-video-to-gif(11)

### CODE REVIEW

File: app/src/main/java/com/bignerdranch/android/criminalintent/CrimeRepository.kt [hunk 1]
```diff
@@ -25,8 +26,21 @@ class CrimeRepository private constructor() {
     Room.databaseBuilder(CriminalIntentApplication.context, CrimeDatabase::class.java, DATABASE_NAME).build()
 
   private val crimeDao = database.crimeDao()
+  private val executor = Executors.newSingleThreadExecutor()
 
   fun getCrimes(): LiveData<List<Crime>> = crimeDao.getCrimes()
 
   fun getCrime(id: UUID): LiveData<Crime?> = crimeDao.getCrime(id)
+
+  fun updateCrime(crime: Crime) {
```
Comment (FerrariBruno) [inline app/src/main/java/com/bignerdranch/android/criminalintent/CrimeRepository.kt]: you can use `=`

File: app/src/main/java/com/bignerdranch/android/criminalintent/CrimeRepository.kt [hunk 2]
```diff
@@ -25,8 +26,21 @@ class CrimeRepository private constructor() {
...
+    executor.execute {
+      crimeDao.updateCrime(crime)
+    }
+  }
+
+  fun addCrime(crime: Crime) {
```
Comment (FerrariBruno) [inline app/src/main/java/com/bignerdranch/android/criminalintent/CrimeRepository.kt]: same

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt [hunk 3]
```diff
@@ -6,19 +6,38 @@ import android.widget.Button
...
 import com.bignerdranch.android.criminalintent.R
 import com.bignerdranch.android.criminalintent.models.Crime
+import com.bignerdranch.android.criminalintent.models.CrimeDetailViewModel
 import com.bignerdranch.android.criminalintent.views.utils.BaseTextWatcher
+import java.util.UUID
 
 class CrimeFragment : Fragment(R.layout.fragment_crime) {
+  companion object {
+    private const val ARG_CRIME_ID = "crime_id"
+    fun newInstance(crimeId: UUID): CrimeFragment {
```
Comment (FerrariBruno) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt:21]:
```suggestion

    fun newInstance(crimeId: UUID): CrimeFragment {
```

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt [hunk 4]
```diff
@@ -6,19 +6,38 @@ import android.widget.Button
...
+      }
+      return CrimeFragment().apply { arguments = args }
+    }
+  }
+
   private lateinit var crime: Crime
   private lateinit var titleField: EditText
   private lateinit var dateButton: Button
   private lateinit var solvedCheckBox: CheckBox
+  private val crimeDeatilViewModel: CrimeDetailViewModel by lazy {
```
Comment (FerrariBruno) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt:34]:
```suggestion
  
  private val crimeDeatilViewModel: CrimeDetailViewModel by lazy {
```

File: app/src/main/java/com/bignerdranch/android/criminalintent/models/CrimeDetailViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  var crimeLiveData: LiveData<Crime?> =
+    Transformations.switchMap(crimeIdLiveData) { crimeId ->
+      crimeRepository.getCrime(crimeId)
+    }
+
+  fun loadCrime(crimeId: UUID) {
+    crimeIdLiveData.value = crimeId
+  }
+
+  fun saveCrime(crime: Crime) {
```
Comment (FerrariBruno) [inline app/src/main/java/com/bignerdranch/android/criminalintent/models/CrimeDetailViewModel.kt]: you can use `=`

**Review by FerrariBruno — APPROVED** [review]

File: app/src/main/java/com/bignerdranch/android/criminalintent/views/MainActivity.kt [hunk 6]
```diff
@@ -3,18 +3,27 @@ package com.bignerdranch.android.criminalintent.views
...
+    val currentFragment = supportFragmentManager.findFragmentById(R.id.fragmentContainer)
+    if (currentFragment == null) {
+      val fragment = CrimeListFragment.newInstance()
+      supportFragmentManager.beginTransaction().add(R.id.fragmentContainer, fragment).commit()
     }
+  }
+
+  override fun openCrimeDetailsView(crimeId: UUID) {
+    val fragment = CrimeFragment.newInstance(crimeId)
+    supportFragmentManager.beginTransaction().replace(R.id.fragmentContainer, fragment)
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/views/MainActivity.kt]: ml

File: app/src/main/java/com/bignerdranch/android/criminalintent/views/MainActivity.kt [hunk 7]
```diff
@@ -3,18 +3,27 @@ package com.bignerdranch.android.criminalintent.views
...
 
-        val currentFragment = supportFragmentManager.findFragmentById(R.id.fragmentContainer)
-        if (currentFragment == null) {
-            val fragment = CrimeListFragment.newInstance()
-            supportFragmentManager.beginTransaction().add(R.id.fragmentContainer, fragment).commit()
-        }
+    val currentFragment = supportFragmentManager.findFragmentById(R.id.fragmentContainer)
+    if (currentFragment == null) {
+      val fragment = CrimeListFragment.newInstance()
+      supportFragmentManager.beginTransaction().add(R.id.fragmentContainer, fragment).commit()
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/views/MainActivity.kt]: ml

**Review by mirland — APPROVED** [review]

**Pull request merged**